rocAL/rocAL/include/meta_data/meta_data.h

```diff
@@ -41,110 +41,144 @@ typedef struct BoundingBoxCord_
 typedef  struct { double xc; double yc; double w; double h; } BoundingBoxCord_xcycwh;
 typedef  std::vector<BoundingBoxCord> BoundingBoxCords;
 typedef  std::vector<BoundingBoxCord_xcycwh> BoundingBoxCords_xcycwh;
-typedef  std::vector<int> BoundingBoxLabels;
-typedef  struct { int w; int h; int original_id; } ImgSize;
+typedef  std::vector<int> Labels;
+typedef  struct { int w; int h; } ImgSize;
 typedef  std::vector<ImgSize> ImgSizes;
 
+typedef struct MetaDataInfo {
+public:
+    uint _img_id = -1;
+    std::string _img_name = "";
+    ImgSize _img_size = {};
+} MetaDataInfo;
 
 struct MetaData
 {
-    int& get_label() { return _label_id; }
-    BoundingBoxCords& get_bb_cords() { return _bb_cords; }
-    BoundingBoxCords_xcycwh& get_bb_cords_xcycwh() { return _bb_cords_xcycwh; }
-    BoundingBoxLabels& get_bb_labels() { return _bb_label_ids; }
-    void set_bb_labels(BoundingBoxLabels bb_label_ids)
-    {
-        _bb_label_ids = std::move(bb_label_ids);
-        _object_count = _bb_label_ids.size();
-    }
-    ImgSize& get_img_size() {return _img_size; }
-    int get_object_count() { return _object_count; }
-    std::vector<size_t> get_bb_label_dims()
-    {
-        _bb_labels_dims = {_bb_label_ids.size()};
-        return _bb_labels_dims;
-    }
-    std::vector<size_t> get_bb_cords_dims()
-    {
-        _bb_coords_dims = {_bb_cords.size(), 4};
-        return _bb_coords_dims;
-    }
-protected:
-    BoundingBoxCords _bb_cords = {}; // For bb use
-    BoundingBoxCords_xcycwh _bb_cords_xcycwh = {}; // For bb use
-    BoundingBoxLabels _bb_label_ids = {};// For bb use
-    ImgSize _img_size = {};
-    int _label_id = -1; // For label use only
-    std::vector<size_t> _bb_labels_dims = {};
-    std::vector<size_t> _bb_coords_dims = {};
-    int _object_count = 0;
+    virtual std::vector<int>& get_label() { };
+    virtual int get_object_count() { };
```

Comment:
Is this wrt bbox? 
Then pls add bbox object count.

Reply:
get_object_count() is not required